### ISSUE TITLE
ASGScalingProcesses registered as 'asg_scheduled_actions', expected to be 'asg_scaling_processes'

### ISSUE THREAD

**Issue by sustacek** [Sceptre/sceptre#841]
I noticed that the default hook for ASG processes is registered as `            "asg_scheduled_actions ="` and not as `asg_scaling_processes`, see https://github.com/Sceptre/sceptre/blob/b7585fd7bf9607230f9d4324166b49cd59cab3ee/setup.py#L59

In the docs (2.2.1) and also the code (https://github.com/Sceptre/sceptre/blob/9a36309ed1daa15ef65504fa550adf161e6d7c6f/sceptre/hooks/asg_scaling_processes.py#L39-L40) it's referenced as `asg_scaling_processes`, but using `!asg_scaling_processes` in `hooks` results in:
```
yaml.constructor.ConstructorError: could not determine a constructor for the tag '!asg_scaling_processes'
  in "<unicode string>", line 40, column 7:
        - !asg_scaling_processes 'suspend: ...
          ^
```

**Comment by ngfgrant** [comment]
Thanks this looks like a documentation error. Fix will be to update the docs

**Comment by reiSh6phoo9o** [comment]
This is still an issue, would be great to get the documentation fixed. Can I contribute somehow?

**Comment by jragland** [comment]
@ngfgrant I am running into this problem currently. 

Sceptre version:
```
~/_/aws-infrastructure on   xx-xxxxx *1 !2 ❯ sceptre --version                                        at  07:49:35
Sceptre, version 3.1.0
```

I have found that if I remove the offending `!` in my `config` file, I can get `sceptre` to build out a `ChangeSet` for me. However, when I go to run said `ChangeSet`, CloudFormation comes back and says it's received an invalid request and doesnt recognize the ARN of the resources im looking for. 

Is this something that is resolved by later versions of `sceptre`?

**Comment by iainelder** [comment]
These commits show the important hook history. The comment is my interpretation.

| commit date | commit hash | comment                                                                  |
|-------------|-------------|--------------------------------------------------------------------------|
| 2017-01-31  | 70dfb57     | Start with the asg_scheduled_actions hook                                |
| 2017-03-30  | d5008a4     | Add the asg_scaling_processes hook                                       |
| 2017-05-15  | 3ecf9cc     | Document both hooks (Today's docs haven't changed)                       |
| 2017-10-11  | e52908b     | Deprecate asg_scheduled_actions in favor of asg_scaling_processes (#216) |
| 2017-10-16  | d300d20     | Document deprecation as a feature of version 1.3.0                       |
| 2017-11-29  | c995826     | Remove asg_scheduled_actions from docs and tests                         |
| 2017-11-29  | c814e6a     | Delete asg_scheduled_actions module and tests                            |
| 2017-11-29  | ec06452     | Register hooks using entry points (#205) (More on this below)            |
| 2023-04-20  | ac9e439     | Move entry points from setup.py to pyproject.toml                        |

The authors intended to replace asg_scheduled_actions with asg_scaling_processes.

But commit ec06452 introduced this bug by retaining the asg_scheduled_actions name for the entry point.

Sceptre would have avoided the confusion by naming the entry point asg_scaling_processes.

This major version of Sceptre needs to keep supporting the old name.

The next minor version can deprecate the old name and add the documented name as another entry point for the same hook.

The next major version can remove the old name.

**Comment by iainelder** [comment]
@jragland , sorry, I didn't understand your point about removing the `!` to generate a change set.

Try again with the latest Sceptre version and if you get an error different from the case originally described you can provide a repro in a new issue.

The OP's point is that the documented name (`asg_scheduled_actions`) doesn't match the real hook name (`asg_scaling_processes`).

That will be fixed in the next major version when #1374 gets merged.
